yellow-paper/docs/decentralisation/block_production.md

```diff
@@ -0,0 +1,384 @@
+:::
+
+```mermaid
+sequenceDiagram
+
+participant Anyone 
+participant Contract as Aztec L1 Contract
+participant Network as Aztec Network
+participant Sequencers
+participant Provers
+
+Anyone ->> Contract: register()
+Anyone --> Anyone: Wait 7 days
+Anyone ->> Network: eligible as a sequencer
+loop Happy Path Block Production
+    Sequencers --> Sequencers: Generate random hashes and rank them
+    Sequencers ->> Contract: Highest ranking sequencers propose blocks
+    Note right of Contract: Proposal phase is over!
+    Contract ->> Network: calculates highest ranking proposal
+    Sequencers ->> Provers: negotiates the cost to prove
+    Sequencers ->> Contract: commits to a proverAddress
+    Provers ->> Contract: proverAddress deposits slashable stake
+    Note right of Network: "preconfirmed" this block is going to be next!
+    Sequencers --> Sequencers: executes public functions
+    Provers --> Provers: generates rollup proofs
+    Provers ->> Contract: submits proofs
+    Contract --> Contract: validates proofs and state transition
+    Note right of Contract: "block confirmed!"
+end 
+Sequencers ->> Contract: exit()
+Sequencers --> Sequencers: wait 7 days 
+```
+
+#### Voting on upgrades
+
+:::danger TODO
+- The flag would probably be something like `proposal|vote` where proposal is a id and vote is `yes|no|abstain`. 
+:::
+
+In the initial implementation of Aztec, sequencers may vote on upgrades alongside block proposals. If they wish to vote alongside an upgrade, they signal by updating their client software or an environment configuration variable. If they wish to vote no or abstain, they do nothing. Because the "election" is randomized, the voting acts as a random sampling throughout the current sequencer set. This implies that the specific duration of the vote must be sufficiently long and RANDAO sufficiently randomized to ensure that the sampling is reasonably distributed.
+
+```mermaid
+sequenceDiagram
+
+participant Contract as Aztec L1 Contract
+participant Network as Aztec Network
+participant Sequencers
+participant Provers
+
+loop Happy Path Block Production
+    Sequencers --> Sequencers: Generate random hashes and rank them
+    Sequencers ->> Contract: Propose block + flag that they desire to upgrade
+    Note right of Contract: Proposal phase is over!
+    Contract ->> Network: calculates highest ranking proposal + vote
+    Sequencers ->> Provers: negotiates the cost to prove
+    Sequencers ->> Contract: commits to a proverAddress
+    Provers ->> Contract: proverAddress deposits slashable stake
+    Note right of Network: "preconfirmed" this block is going to be next!
+    Sequencers --> Sequencers: executes public functions
+    Provers --> Provers: generates rollup proofs
+    Provers ->> Contract: submits proofs
+    Contract --> Contract: validates proofs and state transition
+    Note right of Contract: "block confirmed! votes counted for upgrade!"
+end 
+```
+
+#### Backup mode
```

Comment:
Probably just want to settle on either 'backstop' (as further up the doc) or 'backup'.